compiler/src/main/scala/org/scalajs/core/compiler/GenJSCode.scala

```diff
@@ -3811,28 +3805,16 @@ abstract class GenJSCode extends plugins.PluginComponent
         }
 
         /* Look for a matching method in String. If we find one, we must
-         * redirect it to RuntimeString.
+         * call it without reflective proxy.
          */
         val methodInStringClass = matchingSymIn(StringClass)
         if (methodInStringClass != NoSymbol && methodInStringClass.isPublic) {
           callStatement = js.If(genIsInstanceOf(callTrg, StringClass.tpe), {
-            if (methodInStringClass.owner == ObjectClass) {
-              // If the method is defined on Object, we can call it normally.
-              genApplyMethod(callTrg, methodInStringClass, arguments)
-            } else {
-              // Otherwise, we need to call it on RuntimeString
-              val (rtModuleClass, methodIdent) =
-                encodeRTStringMethodSym(methodInStringClass)
-              val retTpe = methodInStringClass.tpe.resultType
-              val castCallTrg = fromAny(callTrg, StringClass.toTypeConstructor)
-              boxIfNeeded(
-                  genApplyMethod(
-                      genLoadModule(rtModuleClass),
-                      methodIdent,
-                      castCallTrg :: arguments,
-                      toIRType(retTpe)),
-                  retTpe)
-            }
+            val retTpe = methodInStringClass.tpe.resultType
+            val castCallTrg = fromAny(callTrg, StringClass.toTypeConstructor)
+            boxIfNeeded(
+                genApplyMethod(castCallTrg, methodInStringClass, arguments),
+                retTpe)
```

Comment:
Why is string any different here than any other hijacked class?

Reply:
It isn't (apart from the fact that other hijacked classes also have an associated primitive type). There is indeed some factoring out to do here.